### DISCUSSION TITLE
Where rule names for stylecop used in editorconfig comes from?

### DISCUSSION THREAD

**Discussion by JadaVonRuth** [RehanSaeed/EditorConfig#22] [Q&A]
### Where rule names for stylecop used in editorconfig comes from?

Its is not clear how to turn on and off other rules from stylecop analyzer, some of the rules from stylecop are used in this config file but I have no idea where you have got the names and category for it. It is not stated anywhere in stylecop repo.

eg: `dotnet_naming_rule.stylecop_local_fields_must_be_camel_case_rule.severity = silent`

I would like to be able turn off / set severity for other rules from stylecop. Without the nuget of stylecop installed these stylecop rules are not working but when installed it adds tons of other unwanted warnings. There does not seem be a way without adding stylecop json and ruleset config files which kinda is loosing the point of using editorconfig. Is limited only to these rules or there is a way how to determine configuration rule for any stylecop rule in editorconfig so I can get rid of json and ruleset config files?

**Comment by RehanSaeed** [comment]
These are made up names specifically for this file. They don't come from StyleCop. This project replicates the rules in StyleCop but since editorconfig doesn't know anything about StyleCop we have to give the rule a name which gets displayed to developers.
